### ISSUE TITLE
Bouncing on slime blocks below carpets

### ISSUE THREAD

**Issue by minoneer** [NoCheatPlus/Issues#64]
When jumping on a slime block which is covered by a carpet, the player usually bounces back up as if it were a blank slime block. However, on our server it only works for players in creative mode or with nocheatplus.* permissions in survival. Therefore I assume it is NCP blocking the bounce. This is not an issue with bare, uncovered slime blocks.

NoCheatPlus version 3.13.4-BETA-sMD5NET-b901
Spigot version git-Spigot-6c9b0a1-de5c261 (MC: 1.8.8)

**Comment by asofold** [comment]
Reproduced for low-fall-height (also low bounce after bouncing many times with high fall height, but remaining low height). Can you confirm that it seems to be the low-height bouncing?

**Comment by minoneer** [comment]
No, for me it is with high fall hights, everything above 1 block. If I stand on the block and start jumpting, it works as it should. If I jump from a hight of 1 block, I also bounce as I should. Starting from a hight of 2 blocks and up, I just land on the block without any bouncing.
I turned on the logging, it logs a MOVING_SURVIVALFLY (see attachments)

![2015-12-31_11 28 01](https://cloud.githubusercontent.com/assets/6442649/12063976/0e2622b4-afb8-11e5-89d7-655f58bc8b92.png)
![2015-12-31_11 28 57](https://cloud.githubusercontent.com/assets/6442649/12063978/0e274edc-afb8-11e5-8fd0-0180d6d28cc8.png)
![2015-12-31_11 30 10](https://cloud.githubusercontent.com/assets/6442649/12063977/0e26b83c-afb8-11e5-9287-004dffd80b18.png)

**Comment by asofold** [comment]
On second testing run, i have it on the first bounce back from arbitrary height too. I had been testing with an altered version (roughly 908), though.

Looks like we can fix it with the new past move tracking thing.

**Comment by asofold** [comment]
Build 911 should fix or at least reduce the issue (only shortly tested).

**Comment by asofold** [comment]
Looks good with all carpets (build 913), seems that partial carpet coverage still can lead to false positivies (fix following).

**Comment by minoneer** [comment]
I did a quick test with build 913 - seems to work as it should!

Not sure about the partial coverage, but platforms completely covered with carpets are bouncing again.

Thanks for the fast fix!
